### PR TITLE
EVP: Small refactor of keymgmt library code

### PR DESCRIPTION
**Pull request by levitte**
Some functions went directly on keymgmt function pointers instead of
going through the internal KEYMGMT API, which makes for a confusing
read.

Related to #10962

### CODE REVIEW

**Review by mattcaswell — APPROVED** [review]

File: crypto/evp/keymgmt_lib.c [hunk 1]
```diff
@@ -92,7 +92,9 @@ void *evp_keymgmt_export_to_provider(EVP_PKEY *pk, EVP_KEYMGMT *keymgmt,
         struct import_data_st import_data;
 
         import_data.importfn =
-            want_domainparams ? keymgmt->importdomparams : keymgmt->importkey;
+            want_domainparams
```
Comment (slontis) [inline crypto/evp/keymgmt_lib.c]: I think once it becomes this messy it should become a if.
Feel free to ignore if you disagree

File: crypto/evp/keymgmt_lib.c [hunk 2]
```diff
@@ -102,10 +104,11 @@ void *evp_keymgmt_export_to_provider(EVP_PKEY *pk, EVP_KEYMGMT *keymgmt,
             return NULL;
 
         for (j = 0; j < i && pk->pkeys[j].keymgmt != NULL; j++) {
-            int (*exportfn)(void *provctx, OSSL_CALLBACK *cb, void *cbarg) =
+            int (*exportfn)(const EVP_KEYMGMT *keymgmt, void *provdata,
```
Comment (slontis) [inline crypto/evp/keymgmt_lib.c]: A typedef is probably better.. especially if the function signature is going to change. (Again feel free to disagree)

Comment (levitte) [reply crypto/evp/keymgmt_lib.c]: I would if it was going to get used in more places, but as far as I'm concerned, this is the only spot where using a function pointer for this particular signature, so...

(don't you wish that C had a `typeof` that you could give a function name, and that would be the signature of that function?  Alas...)

File: crypto/evp/keymgmt_lib.c (hunk 1 above)
Comment (levitte) [reply crypto/evp/keymgmt_lib.c]: I disagree.

**Comment by levitte** [comment]
Merged

658608c471a6e1f9b6d7f88c060a7adb77d7d334 EVP: Small refactor of keymgmt library code

**Pull request closed without merging**